backend/tests/helpers/query_benchmark/data_generator.py

```diff
@@ -0,0 +1,82 @@
+from abc import abstractmethod
+from pathlib import Path
+from typing import Callable
+
+from rich.console import Console
+from rich.progress import Progress
+
+from infrahub.core.node import Node
+from tests.helpers.query_benchmark.db_query_profiler import InfrahubDatabaseProfiler, ProfilerEnabler, query_analyzer
+
+
+class DataGenerator:
+    """
+    Abstract class responsible for loading data into a given database.
+    """
+
+    def __init__(self, db: InfrahubDatabaseProfiler) -> None:
+        self.db = db
+
+    async def init(self) -> None:
+        """
+        Any previous step before loading and profiling data should be implemented here.
+        """
+
+    @abstractmethod
+    async def load_data(self, nb_elements: int) -> None:
+        raise NotImplementedError("Abstract method")
+
+    async def _save_obj(self, obj: Node) -> None:
```

Comment:
where is it being used ?
we shouldn't have to create both a session and a transaction, the transaction will be created automatically if it doesn't exist